src/js/modules/accounts/components/forms/SignUpForm.js

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import { connect } from "react-redux";
+import { Field, reduxForm } from "redux-form";
+import injectSheet from "react-jss";
+import { EMAIL_REGEX } from "../../../../constants";
+
+const styles = {
+  successMessage: {
+    color: "green",
+  },
+  errorMessage: {
+    color: "red",
+  },
+};
+
+const validate = formValues => {
+  const errors = {};
+  if (!formValues.firstName) {
+    errors.firstName = "Required";
+  }
+  if (!formValues.lastName) {
+    errors.lastName = "Required";
+  }
+  if (!formValues.email) {
+    errors.email = "Required";
+  } else if (!EMAIL_REGEX.test(formValues.email)) {
+    errors.email = "Invalid email address";
+  }
+  if (!formValues.password) {
+    errors.password = "Required";
+  } else if (formValues.password.length < 8) {
+    errors.password = "Password is too short (minimum is 8 characters).";
+  }
+  if (formValues.password !== formValues.passwordConfirmation) {
+    errors.passwordConfirmation =
+      "Password and password confirmation do not match.";
+  }
+  return errors;
+};
+
+const renderField = ({
+  input,
+  label,
+  type,
+  meta: { touched, error, warning },
+}) => {
+  return (
+    <div>
+      <label>{label}</label>
+      <div>
+        <input {...input} placeholder={label} type={type} />
+        {touched &&
+          ((error && <span>{error}</span>) ||
+            (warning && <span>{warning}</span>))}
+      </div>
+    </div>
+  );
+};
+
+const SignUpForm = ({ classes, handleSubmit, submitting, status }) => {
```

Comment:
There's a consistent boilerplate for each form. Have you thought about factoring some of it out? It's okay if you can't

Reply:
Both these comments are the same as above, and I've replied to them.

Reply:
No? Read this one again lol. I'm saying that there's a couple things like the styles, the error mapping, etc. that repeat. If you want, you can try to factor them out into a repeatable component

Reply:
Yes, but there are also significant differences (what the inputs are, any special messages, etc.) I think separate components would be better than several `switch`'s in one component.

Reply:
Alright, that's a fair assessment